### PR TITLE
Build vendored libusb with CMake if on Apple or not available system-wide

### PR DESCRIPTION
**Pull request by ligfx**
This fixes issues like https://forums.dolphin-emu.org/Thread-osatomicincrement32barrier-missing-on-osx-10-9-5 .

Currently, the vendored version of libusb is only built when using Visual Studio. With CMake, libusb is included only when already available on the system, which can then get pulled in to a macOS application bundle. If the system version isn't built to be compatible with previous versions of macOS (unlike Dolphin, which is built with `-mmacosx-version-min=10.9`), then it breaks the entire application.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4230)

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by clinchergt** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by ligfx** [comment]
Anecdotal evidence that it does now work as intended on macOS 10.9: https://forums.dolphin-emu.org/Thread-osatomicincrement32barrier-missing-on-osx-10-9-5?pid=423372

File: CMakeLists.txt [hunk 1]
```diff
@@ -630,7 +629,7 @@ endif()
 add_subdirectory(Externals/Bochs_disasm)
 include_directories(Externals/Bochs_disasm)
 
-add_subdirectory(Externals/glslang)
+# add_subdirectory(Externals/glslang)
```
Comment (leoetlino) [inline CMakeLists.txt]: unrelated change

File: Externals/libusb/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    find_library(COREFOUNDATION_LIBRARY CoreFoundation)
+    find_library(IOKIT_LIBRARY IOKit)
+    find_library(OBJC_LIBRARY objc)
+    target_link_libraries(usb PRIVATE
+        ${COREFOUNDATION_LIBRARY}
+        ${IOKIT_LIBRARY}
+        ${OBJC_LIBRARY}
+    )
+    set(OS_DARWIN TRUE)
+elseif(ANDROID)
```
Comment (leoetlino) [inline Externals/libusb/CMakeLists.txt]: Dolphin on Android doesn't use libusb.

File: CMakeLists.txt (hunk 1 above)
Comment (ligfx) [inline CMakeLists.txt]: Done.

File: Externals/libusb/CMakeLists.txt (hunk 2 above)
Comment (ligfx) [inline Externals/libusb/CMakeLists.txt]: In the spirit of the other Externals' CMakeLists, I've commented out this section of code instead of deleting it. Let me know if you'd like an approach other than that.

**Comment by shuffle2** [comment]
@dolphin-emu-bot rebuild
